nixos/common/networking.nix

```diff
@@ -24,4 +24,9 @@
   systemd.services.systemd-networkd.stopIfChanged = false;
   # Services that are only restarted might be not able to resolve when resolved is stopped before
   systemd.services.systemd-resolved.stopIfChanged = false;
+
+  # If networkd in initrd is enabled, just copy the configuration of the normal networkd.
+  # Potentially can miss some drivers, but hopefully it would gracefully degrade in those cases
+  # See https://github.com/NixOS/nixpkgs/issues/157034#issuecomment-2163512812
+  boot.initrd.systemd.network = lib.mkDefault (builtins.removeAttrs config.systemd.network ["enable"]);
```

Comment:
Also I am unsure if this is a good idea. If for example someone sets up a wireguard tunnel, they will likely miss the keys in initrd. There is probably more examples like that.

Reply:
or when doing bridges/bonding you need extra kernel modules https://gitea.c3d2.de/c3d2/nix-config/src/branch/master/modules/baremetal.nix#L18